src/vulkan/vk-backend.cpp

```
@@ -31,11 +31,17 @@ Result BackendImpl::createDevice(const DeviceDesc& desc, IDevice** outDevice)
 Result BackendImpl::enumerateAdapters()
 {
     VulkanModule module;
-    SLANG_RETURN_ON_FAIL(module.init());
+    if (SLANG_FAILED(module.init()))
+    {
+        return SLANG_FAIL;
+    }
     SLANG_RHI_DEFERRED({ module.destroy(); });
```

Cppcheck (2.20.0)

[style] The function 'push' is never used.

(unusedFunction)